### PR TITLE
rearrange bibdetails types and preprocessing

### PR DESCRIPTION
**Pull request by charmingduchess**
- rename and move SearchResult type to Bib type. I think this will clear up some of the confusion around the types/models. In reality, DAPI returns the same kinds of objects whether its the resources/bibid route or as a search result
- extract bib details types into their own file. They really refer to types that are only used by the bib details components.
- move bib detail preprocessing into the BibDetails class. 
- add owning institution detail to BibDetails' bottom details

### CODE REVIEW

File: src/models/BibDetails.ts [hunk 1]
```diff
@@ -113,6 +117,100 @@ export default class BibDetailsModel {
...
+  getNoteType = (note: Note) => {
+    const type = note.noteType || ""
+    return type.toLowerCase().includes("note") ? type : `${type} (note)`
+  }
+
+  /**
+   * Combines properties from matching (i.e. parallel) elements as necessary
+   * Right now, this is only needed to add the 'noteType' in case of parallel notes
+   */
+  combineMatching(primaryValue: string, parallelValue: string | Note) {
```
Comment (dgcohen) [inline src/models/BibDetails.ts:155]: Small nit, but I'm thinking combineMatching, interleaveParallel and matchParallel's could use more descriptive fields? Even if it's just interleaveParallelFields etc?

**Comment by dgcohen** [comment]
Thanks so much for this! This looks way better. Just one small suggestion about naming but otherwise I think this is a big improvement.

**Comment by charmingduchess** [comment]
@dgcohen good 2 go?

**Review by dgcohen — APPROVED** [review]
Looks great! Thanks

**Pull request merged**